provider.go

```diff
@@ -287,16 +288,42 @@ func (g *InstanceGroup) Increase(
 	ctx context.Context,
 	delta int,
 ) (succeeded int, err error) {
-	servers := make([]*cloudscale.Server, 0, delta)
-	errs := make([]error, 0)
-
 	publicKey, err := g.publicKey()
 	if err != nil {
 		return 0, err
 	}
 
 	tagMap := g.tagMap()
 
+	// Channel for server uuids
+	servers := make(chan string)
+	// Because concurency, we use a channel for collecting errors, as
+	// slice.append is not thread safe
+	errors_c := make(chan error)
```

Comment:
You might want to consider using `sync.Mutex` instead, as I think it might make the implementation a bit more straight-forward. That being said, if you keep the channel, you have to close it, or the `range errors_c` expression below will never finish.

Reply:
you're right, that feels easier